polus-label-to-vector-plugin/src/main.py

```diff
@@ -1,190 +1,378 @@
+                tile[numpy.newaxis, numpy.newaxis, 0:reader.Z,
+                     y_overlap:y_max - y_min + y_overlap,
+                     x_overlap:x_max - x_min + x_overlap]
+            ).astype(numpy.float32)
+    except Exception as e:
+        logging.error(f'failed on {input_path.name} with error {e}')
+        raise e
+
+    return True
+
+
+def flow_thread_2d(
+        input_path: Path,
+        zarr_path: Path,
+        use_gpu: bool,
+        device: torch.device,
+        coordinates: tuple[int, int, int],
+) -> bool:
+    """ Converts labels to flows.
+
+    This function converts labels in each tile to vector field.
 
-    root = zarr.open(str(zfile))[0]
-    
-    with BioReader(input_path) as br:
+    Args:
+        input_path: Path of input image collection.
+        zarr_path: Path where output zarr file will be saved.
+        use_gpu: Path where output zarr file will be saved.
+        device: Path where output zarr file will be saved.
+        coordinates: (x, y, z) coordinates of the tile in the image.
+    """
+
+    logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                        datefmt='%d-%b-%y %H:%M:%S')
+    flow_logger = logging.getLogger("flow_2d")
+    flow_logger.setLevel(logging.INFO)
+
+    # noinspection PyTypeChecker
+    root = zarr.open(str(zarr_path))[0]
+    x, y, z = coordinates
+
+    with BioReader(input_path) as reader:
         x_min = max([0, x - TILE_OVERLAP])
-        x_max = min([br.X, x + TILE_SIZE + TILE_OVERLAP])
+        x_max = min([reader.X, x + TILE_SIZE + TILE_OVERLAP])
         y_min = max([0, y - TILE_OVERLAP])
-        y_max = min([br.Y, y + TILE_SIZE + TILE_OVERLAP])
-        
-        # Normalize
-        I = br[y_min:y_max, x_min:x_max, z:z+1, 0, 0].squeeze()
-        _, image = np.unique(I, return_inverse=True)
-        image = image.reshape(y_max-y_min,x_max-x_min)
-
-        flow = dynamics.masks_to_flows(image,
-                                       use_gpu,
-                                       dev)[0]
-        
-        logger.debug('Computed flows on slice %d tile(y,x) %d:%d %d:%d ', z, y,
-                     y_max, x, x_max)
-        flow_final = flow[:, :, :, np.newaxis, np.newaxis].transpose(1, 2, 3, 0, 4)
+        y_max = min([reader.Y, y + TILE_SIZE + TILE_OVERLAP])
+
+        tile = reader[y_min:y_max, x_min:x_max, z:z + 1, 0, 0].squeeze()
+
+        if not numpy.any(tile):
+            flow_logger.debug(f'got tile with no objects in file {input_path.name}')
+            flow_shape = (y_max - y_min, x_max - x_min, reader.Z, 2, 1)
+            flow = numpy.zeros(flow_shape, dtype=numpy.float32)
+        else:
+            # Normalize
+            labels, masks = numpy.unique(tile, return_inverse=True)
+            if len(labels) == 1:
+                flow_logger.debug(f'got tile with only one object in file {input_path.name}')
+                masks += 1
```

Comment:
Same comment as above, this appears to be nearly identical to what `clean_3d_flow` does.